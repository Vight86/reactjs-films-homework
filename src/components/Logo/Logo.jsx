import React from 'react';
import propTypes from 'prop-types';
import styles from './Logo.scss';

const Logo = ({ className, children }) => (
  <p className={styles[className]}>
    <a href="/" title="Films">
      {children}
    </a>
  </p>
);

Logo.propTypes = {
  className: propTypes.string.isRequired,
  children: propTypes.string.isRequired,
};

export default Logo;
