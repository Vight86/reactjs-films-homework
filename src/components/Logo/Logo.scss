@import '../../utils/variables';

.logo {
  margin: 0;
  font-family: $font-secondary;
  font-size: 5em;
  letter-spacing: 0.04em;

  & > a {
    text-decoration: none;
  }
}

.primary {
  @extend .logo;

  & > a {
    color: $color-light;
  }
}

.secondary {
  @extend .logo;

  & > a {
    color: $color-secondary;
  }
}
