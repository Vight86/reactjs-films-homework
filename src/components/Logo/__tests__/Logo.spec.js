import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Logo from '../index';

let renderer;
let result;

beforeEach(() => {
  renderer = new ShallowRenderer();
  renderer.render(<Logo className="test">Title</Logo>);
  result = renderer.getRenderOutput();
});

afterEach(() => {
  renderer = null;
  result = null;
});

describe('render Logo component', () => {
  it('render correctly', () => {
    expect(result).toMatchSnapshot();
  });
});
