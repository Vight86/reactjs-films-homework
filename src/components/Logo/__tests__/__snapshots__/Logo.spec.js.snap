// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Logo component render with props 1`] = `
<p
  className="test"
>
  <a
    href="/"
    title="Films"
  >
    Films
  </a>
</p>
`;

exports[`render Logo component render without props 1`] = `
<p
  className="primary"
>
  <a
    href="/"
    title="Films"
  >
    Films
  </a>
</p>
`;
